enterprise_subsidy/apps/api/v1/views/transaction.py

```diff
@@ -120,19 +148,29 @@ def requested_transaction_uuid(self):
 
         For detail endpoints, the PK can simply be found in ``self.kwargs``.
         """
-        return self.kwargs.get("uuid")
+        transaction_uuid = self.kwargs.get("uuid")
+        if transaction_uuid:
+            try:
+                transaction_uuid = UUID(transaction_uuid)
+            except ValueError as exc:
+                raise ParseError(f"{transaction_uuid} is not a valid uuid.") from exc
+        return transaction_uuid
 
     @property
     def requested_subsidy_uuid(self):
         """
-        Fetch the subsidy UUID from either the URL query parameters or JSON body.
+        Fetch the subsidy UUID from either the URL query parameters or request body.
+
+        Note: if the request contains both, the query parameter takes priority.
         """
         subsidy_uuid = self.request.query_params.get("subsidy_uuid")
         if not subsidy_uuid:
+            subsidy_uuid = self.request.data.get("subsidy_uuid")
+        if subsidy_uuid:
             try:
-                subsidy_uuid = json.loads(self.request.body).get("subsidy_uuid")
-            except json.decoder.JSONDecodeError:
-                pass
+                subsidy_uuid = UUID(subsidy_uuid)
+            except ValueError as exc:
+                raise ParseError(f"{subsidy_uuid} is not a valid uuid.") from exc
```

Comment:
Part of this is simplification to avoid manually decoding JSON when `self.request.data` already contains the decoded JSON.  The other part of this is to actually throw a ParseError which leads to a 404 instead of letting some other code path throw a 500 error.